compiler/ast/trees.nim

```diff
@@ -97,7 +97,7 @@ makeTreeEquivalenceProc(exprStructuralEquivalentStrictSymAndComm,
   relaxedKindCheck = false,
   symCheck     = a.sym == b.sym,
   floatCheck   = sameFloatIgnoreNan(a.floatVal, b.floatVal),
-  typeCheck    = true,
+  typeCheck    = a.typ == b.typ,
```

Comment:
Doesn't this make type comparison apply for all nodes?

Reply:
Perhaps, I tend to think so but am not entirely sure yet.Though for nkType it only makes sense either to check it's type way.

Reply:
I think we should ensure that it only checks types for nodes where the type stands in for what would otherwise be literal AST. An example of which would be `nlType`